File: Daily Challenge/Python/Bottom Left To Top Right Diagonals Pattern 13.9 2025.py
N = int(input())
hyp = [['-' for _ in range(N * 2)] for _ in range(N * 2)]
for i in range(0, N * 2, N):
    for j in range(0, N * 2, N):
        row = i + N - 1
        col = j
        cnt = 0
        while cnt < N:
            hyp[row][col] = '*'
            row -= 1
            col += 1
            cnt += 1
for i in hyp:
    print(*i, sep='')

File: Daily Challenge/C/Bottom Left To Top Right Diagonals Pattern 13.9.2025.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);
    int size = N * 2;
    char hyp[size][size];

    for (int i = 0; i < size; i++)
        for (int j = 0; j < size; j++)
            hyp[i][j] = '-';

    for (int i = 0; i < size; i += N) {
        for (int j = 0; j < size; j += N) {
            int row = i + N - 1, col = j, cnt = 0;
            while (cnt < N) {
                hyp[row][col] = '*';
                row--;
                col++;
                cnt++;
            }
        }
    }

    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++)
            printf("%c", hyp[i][j]);
        printf("\n");
    }
    return 0;
}